sections/testing/mutation-testing.tex

```diff
@@ -1,2 +1,15 @@
-Testing the test set
 \section{Mutation testing}
+Even with a code coverage of 100\%, the traditional test coverage measures only how much of the code is executed by the tests, and not whether the tests are able to detect faults in the code \cite{pitest}.
+This means that it is possible to get a full line, statement, and branch coverage by simply executing the code without any useful assertions in the test suite.
+One way to combat this problem is with the use of mutation testing.
+The idea behind mutation testing is to run some automated software (the mutation testing system), which will generate mutants by modifying minor parts of the system such as replacing a less than check with a greater than check.
+The objective is to have a test suite that \textit{kills} the mutants by revealing that the mutant behavior differs from the original program \cite{mutationtesting}.
```

Comment:
Synes ikke at det er helt klart at en mutant dør på grund af at vores unit test fejler.
forklar hvordan vi revealer at mutant behaviour differs from the original program.